tests/integration/testNumber.intTest.ts

```diff
@@ -1,49 +1,43 @@
-import {TestNumberService} from "../../src/services/TestNumberService";
-import {TestNumber} from "../../src/models/TestNumber";
+import {TestNumber} from "../../src/models/NumberModel";
 import {generateTestNumber} from "../../src/functions/generateTestNumber";
 import lambdaTester from "lambda-tester";
-import {DynamoDBService} from "../../src/services/DynamoDBService";
+import {emptyDatabase, populateDatabase} from "../util/dbOperations";
 
 describe("POST /test-number", () => {
-    const testNumberService: TestNumberService = new TestNumberService(new DynamoDBService());
     const lambda = lambdaTester(generateTestNumber);
-    const expectedFirstTestNumber: TestNumber = {
-        testNumber: "W01A00128",
-        id: "W01",
-        certLetter: "A",
-        sequenceNumber: "001",
-        testNumberKey: 1
-    };
 
     beforeAll(async () => {
-        // Reset the Database
-         await testNumberService.dbClient.batchDelete([{testNumberKey: 1}]);
-         await testNumberService.dbClient.put(expectedFirstTestNumber);
+        jest.restoreAllMocks();
+        await emptyDatabase();
+    });
+    beforeEach(async () => {
+        await populateDatabase();
+    });
+    afterEach(async () => {
+        await emptyDatabase();
+    });
+    afterAll(async () => {
+        await populateDatabase();
     });
 
     context("when a new test-number is requested when only the seed data is present", () => {
         it("should respond with HTTP 200 and a next valid test number", () => {
             const nextTestNumber: TestNumber = {
-                testNumber: "W01A00229",
+                testNumber: "W01A00128",
                 id: "W01",
                 certLetter: "A",
-                sequenceNumber: "002",
+                sequenceNumber: "001",
                 testNumberKey: 1
             };
             return lambda
-            .expectResolve((response: any) => {
-                expect(response.headers["Access-Control-Allow-Origin"]).toEqual("*");
-                expect(response.headers["Access-Control-Allow-Credentials"]).toEqual(true);
-                expect(response.statusCode).toEqual(200);
-                expect(nextTestNumber).toEqual(JSON.parse(response.body));
-            });
+                .expectResolve((response: any) => {
+                    expect(response.headers["Access-Control-Allow-Origin"]).toEqual("*");
```

Comment:
would  recommend use of expect.assertions(4) (or 5?) before the  return in order to ensure that all your assertions actually get called. I'm  not completely familiar with expectResolve's functionality, but just in case it fails silently.

Reply:
done, expectResolve doesn't count as an assertion, so 4 assertions :D